src/components/CallView/CallView.vue

```diff
@@ -1,75 +1,310 @@
+		},
+
+		screenSharingActive() {
+			return this.screens.length > 0
 		},
 
-		firstLetterOfGuestName() {
-			const customName = this.displayName !== t('spreed', 'Guest') ? this.displayName : '?'
-			return customName.charAt(0)
+		darkBackground() {
+			return this.remoteParticipantsCount > 0 || this.screenSharingActive
 		},
+
 	},
-}
-</script>
 
-<style lang="scss" scoped>
-.stage {
-	background: #000000;
-	width: 100%;
-	height: 100%;
+	watch: {
+
+		'darkBackground': function(darkBackground) {
+			this.$emit('hasDarkBackground', darkBackground)
+		},
+
+		'localMediaModel.attributes.localScreen': function(localScreen) {
+			this._setScreenAvailable(this.localCallParticipantModel.attributes.peerId, localScreen)
+		},
 
-	.promoted-stream {
-		height: auto;
-	}
+		'callParticipantCollection.callParticipantModels': function(models) {
+			const addedModels = models.filter(model => !this.sharedDatas[model.attributes.peerId])
+			const removedModelIds = Object.keys(this.sharedDatas).filter(sharedDataId => models.find(model => model.attributes.peerId === sharedDataId) === undefined)
 
-	.video-container-row {
-		height: 180px;
-		position: absolute;
-		bottom: 0;
-		width: 100%;
+			removedModelIds.forEach(removedModelId => {
+				this.$delete(this.sharedDatas, removedModelId)
+
+				this.speakingUnwatchers[removedModelId]()
+				// Not reactive, but not a problem
+				delete this.speakingUnwatchers[removedModelId]
+
+				this.screenUnwatchers[removedModelId]()
+				// Not reactive, but not a problem
+				delete this.screenUnwatchers[removedModelId]
+
+				const index = this.speakers.findIndex(speaker => speaker.id === removedModelId)
+				this.speakers.splice(index, 1)
+
+				this._setScreenAvailable(removedModelId, false)
+			})
+
+			addedModels.forEach(addedModel => {
+				const sharedData = {
+					promoted: false,
+					videoEnabled: true,
+					screenVisible: false,
+				}
+
+				this.$set(this.sharedDatas, addedModel.attributes.peerId, sharedData)
+
+				// Not reactive, but not a problem
+				this.speakingUnwatchers[addedModel.attributes.peerId] = this.$watch(function() {
+					return addedModel.attributes.speaking
+				}, function(speaking) {
+					this._setSpeaking(addedModel.attributes.peerId, speaking)
+				})
+
+				this.speakers.push({
+					id: addedModel.attributes.peerId,
+					active: false,
+				})
+
+				// Not reactive, but not a problem
+				this.screenUnwatchers[addedModel.attributes.peerId] = this.$watch(function() {
+					return addedModel.attributes.screen
+				}, function(screen) {
+					this._setScreenAvailable(addedModel.attributes.peerId, screen)
+				})
+			})
+		},
+
+		'speakers': function() {
+			this._setPromotedParticipant()
+		},
+
+		'screenSharingActive': function() {
+			this._setPromotedParticipant()
+		},
+
+		'screens': function() {
+			this._setScreenVisible()
+		},
+
+	},
+
+	mounted() {
+		this.localMediaModel = new LocalMediaModel()
```

Comment:
This is not working. Somehow the watchers are active before everything else. It can only be fixed by moving the initialisation to beforeCreate, but then `this.` is not set later on in the watchers when they use computeds...

@skjnldsv @ma12-co had a similar problem before?

Reply:
What does "it is not working" means?

Reply:
Using callParticipantsModels on null, etc.

Reply:
Why not initializing it in the data directly?

Reply:
Thats when you should stop working.... right?

Reply:
I don't know, I don't really understand what everything is doing here :p

Reply:
Yeah, I ment if you dont have the idea anymore to create the thing in the data directly, one should take a break